### PR TITLE
stricter checks for login validation

### PR DESCRIPTION
**Pull request by waghswapnil**
- stricter checks
- allow login for S-series IMC servers
Signed-off-by: Swapnil Wagh <waghswapnil@gmail.com>

### CODE REVIEW

File: imcsdk/imcsession.py [hunk 1]
```diff
@@ -461,32 +461,43 @@ def _validate_connection(self):
...
-                        model_name in valid_models):
-                    self._logout()
-                    return False
+        request = config_resolve_class(cookie=self.__cookie,
+                                       class_id="biosUnit")
+        response = self.post_elem(request)
+
+        if not response or response.error_code != 0 or \
+                len(response.out_configs.child) == 0:
+            return False
```
Comment (vvb) [inline imcsdk/imcsession.py:492]: should you be logging out here before returning?

Comment (waghswapnil) [reply imcsdk/imcsession.py:492]: yes, good catch ... will take care of it

**Review by vvb — CHANGES_REQUESTED** [review]
LGTM, other than the comment

**Review by vvb — APPROVED** [review]
LGTM

**Pull request merged**